src/scripts/main.js

```diff
@@ -1,3 +1,51 @@
+/* eslint-disable max-len */
 'use strict';
 
-// write code here
+const list = document.querySelector('ul');
+
+const helperFn = {
+  sort(prev, next) {
+    return helperFn.convertToNumber(next.dataset.salary) - helperFn.convertToNumber(prev.dataset.salary);
+  },
+  convertToNumber(number) {
+    return +number.replace('$', '').replace(',', '.');
+  },
+  appendToList(item) {
+    list.innerHTML += `
+      <li
+        data-position="${item.dataset.position}"
+        data-salary="${item.dataset.salary}"
+        data-age="${item.dataset.age}"
+      >
+      ${item.innerText}
+      </li>
+    `;
+  },
+};
+
+function sortList(array) {
```

Comment:
The name of this parameter is incorrect as later you are passing here a single ul node rather than array of nodes. Rename it or adjust passed value.